### PR TITLE
CU-86du7e4k8 BSEthereum - Implement new BlockchainDataService for Neo…

### PR DESCRIPTION
**Pull request by lopescode**
…X using Blockscout

### CODE REVIEW

File: packages/bs-ethereum/src/BlockscoutNeoXBDSEthereum.ts [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+          next_page_params: params.nextPageParams,
+        },
+      }
+    )
+
+    const nativeToken = BSEthereumHelper.getNativeAsset(this._network)
+
+    const transactions: TransactionResponse[] = []
+
+    const promises = data.items.map(async item => {
```
Comment (raulduartep) [inline packages/bs-ethereum/src/BlockscoutNeoXBDSEthereum.ts:160]: Is it necessary to be async ?

File: packages/bs-ethereum/src/__tests__/BlockscoutNeoXBDSEthereum.spec.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  TransactionsByAddressResponse,
+  TransactionTransferAsset,
+  TransactionTransferNft,
+} from '@cityofzion/blockchain-service'
+import { BlockscoutNeoXBDSEthereum } from '../BlockscoutNeoXBDSEthereum'
+import { BSEthereumHelper } from '../BSEthereumHelper'
+
+const network = BSEthereumHelper.TESTNET_NETWORKS.find(network => network.id === '12227332')!
+
+// TODO: When all test are finished import the address 0xD81a8F3c3f8b006Ef1ae4a2Fd28699AD7E3e21C5 on NEON to test the interface
```
Comment (raulduartep) [inline packages/bs-ethereum/src/__tests__/BlockscoutNeoXBDSEthereum.spec.ts:13]: Remove it

File: packages/bs-ethereum/src/__tests__/BlockscoutNeoXBDSEthereum.spec.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+    const blockscoutBDSNeoX = new BlockscoutNeoXBDSEthereum(network)
+    const transaction = await blockscoutBDSNeoX.getTransactionsByAddress({ address })
+
+    expect(transaction).toEqual(expectedResponse)
+  })
+
+  // This test is being skipped because we don't have any address with more than 50 transactions yet using Dora blockscout endpoint in testnet
+  // Change the endpoint to see the difference: https://xt3scan.ngd.network/api/v2/addresses/0xdc0b6d0F38738a89BA9193B50fF4111030f0d329/transactions
+  it.skip('Should return transfer transactions by address with next page', async () => {})
```
Comment (raulduartep) [inline packages/bs-ethereum/src/__tests__/BlockscoutNeoXBDSEthereum.spec.ts:131]: Remove this test if there is no way to test

File: packages/bs-ethereum/src/__tests__/BlockscoutNeoXEDSEthereum.spec.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          name: 'Wrapped GAS v10',
+          symbol: 'WGAS10',
+        },
+        usdPrice: 0,
+      },
+    ])
+  })
+
+  // This test is skipped because blockscout doesn't support token price history yet
+  it.skip('Should return token price history', () => {})
```
Comment (raulduartep) [inline packages/bs-ethereum/src/__tests__/BlockscoutNeoXEDSEthereum.spec.ts:37]: Remove this test

File: packages/bs-ethereum/src/__tests__/BlockscoutNeoXEDSEthereum.spec.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { GetTokenPricesParams } from '@cityofzion/blockchain-service'
+import { BlockscoutNeoXEDSEthereum } from '../BlockscoutNeoXEDSEthereum'
+import { BSEthereumHelper } from '../BSEthereumHelper'
+
+const network = BSEthereumHelper.TESTNET_NETWORKS.find(network => network.id === '12227332')!
+
+describe('BlockscoutNeoXEDSEthereum', () => {
+  // This test is skipped because blockscout doesn't support token prices in testnet
```
Comment (raulduartep) [inline packages/bs-ethereum/src/__tests__/BlockscoutNeoXEDSEthereum.spec.ts:8]: Why don't you try to use mainnet ?

**Comment by lopescode** [comment]
@raulduartep @melanke DO NOT MERGE THIS!

We're waiting NeoX mainnet blockscout url being deployed

**Pull request merged**